deepchecks/core/presentation/check_result/html.py

```diff
@@ -0,0 +1,166 @@
+from pandas.io.formats.style import Styler
+from plotly.basedatatypes import BaseFigure
+from typing_extensions import Literal
+
+from deepchecks.core.check_result import CheckResult
+from deepchecks.utils.strings import get_docs_summary
+from deepchecks.core.presentation.abc import HtmlSerializer
+from deepchecks.core.presentation.dataframe import DataFramePresentation
+from deepchecks.core.presentation.common import aggregate_conditions, form_output_anchor, form_check_id
+
+
+__all__ = ['CheckResultSerializer']
+
+
+CheckResultSection = t.Union[
+    Literal['condition-table'],
+    Literal['additional-output'],
+]
+
+
+class CheckResultSerializer(HtmlSerializer[CheckResult]):
+
+    def __init__(self, value: CheckResult, **kwargs):
+        self.value = value
+
+    def serialize(
+        self,
+        output_id: t.Optional[str] = None,
+        include: t.Optional[t.Sequence[CheckResultSection]] = None,
+        **kwargs
+    ) -> str:
+        sections_to_include = verify_include_parameter(include)
+        sections = [self.prepare_header(output_id), self.prepare_summary()]
+
+        if 'condition-table' in sections_to_include:
+            sections.append(''.join(self.prepare_conditions_table(output_id=output_id)))
+
+        if 'additional-output' in sections_to_include:
+            sections.append(''.join(self.prepare_additional_output(output_id)))
+        
+        return ''.join(sections)
+    
+    def prepare_header(self, output_id: t.Optional[str] = None) -> str:
+        header = self.value.get_header()
+        header = f'<b>{header}</b>'
+        if output_id is not None:
+            check_id = form_check_id(self.value.check, output_id)
+            return f'<h4 id="{check_id}">{header}</h4>'
+        else:
+            return f'<h4>{header}</h4>'
+    
+    def prepare_summary(self) -> str:
+        summary = get_docs_summary(self.value.check)
+        return f'<p>{summary}</p>'
+    
+    def prepare_conditions_table(
+        self,
+        max_info_len: int = 3000, 
+        include_icon: bool = True,
+        include_check_name: bool = False,
+        output_id: t.Optional[str] = None,
+    ) -> str:
+        table = DataFramePresentation(aggregate_conditions(
+            self.value, 
+            max_info_len=max_info_len,
+            include_icon=include_icon,
+            include_check_name=include_check_name,
+            output_id=output_id
+        )).to_html()
+        return f'<h5>Conditions Summary</h5>{table}'
+    
+    def prepare_additional_output(
+        self,
+        output_id: t.Optional[str] = None
+    ) -> t.List[str]:
+        return DisplayItemsHandler.handle_display(self.value.display)
+
+
+class DisplayItemsHandler:
+
+    @classmethod
+    def handle_display(
+        cls,
+        display: t.List[t.Union[t.Callable, str, pd.DataFrame, Styler]],
+        output_id: t.Optional[str] = None,
+    ) -> t.List[str]:
+        output = [cls.header()]
+
+        for item in display:
```

Comment:
why do all the handles in here transform to HTML? If later I want to convert it to json?

Reply:
the reason why I added this class is to make it possible to reuse HTML serialization functionality in widget serializer without using inheritance which makes code flow harder to follow and understand.

If it is useful with JSON serializer I will adapt it and will make it more flexible.

Reply:
I created a separate protocol called  `ABCDisplayItemsHandler` that is now used everywhere in the subpackage when we need to deal with display items